torch/profiler/strobelight_cli_function_profiler.py

```diff
@@ -0,0 +1,311 @@
+import functools
```

Comment:
Every function should have types because mypy doesn't typecheck functions with missing types.  
If you want to enforce this for the whole file you can add to the first line:
```
# mypy: disallow-untyped-defs
```